main.go

```diff
@@ -2,42 +2,25 @@ package main
 
 import (
 	"fmt"
+	searchPokemon "pykadex-backend/apis/search"
+	uploadPokemon "pykadex-backend/apis/upload"
+	"strconv"
+
 	"log"
 	"net/http"
-	"time"
+
+	"github.com/gorilla/mux"
 )
 
 func main() {
-	m := http.NewServeMux()
-
-	const addr = ":8000"
-
-	m.HandleFunc("/", handleSearch)
-
-	srv := http.Server{
-		Handler: m,
-		Addr: addr,
-		WriteTimeout: 30 * time.Second,
-		ReadTimeout: 30 * time.Second,
-	}
-
-	fmt.Println("server started on port", addr)
-	err := srv.ListenAndServe()
-	log.Fatal(err)
-}
+	r := mux.NewRouter()
 
-func handleSearch(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "text/html")
-	w.Header().Set("Access-Control-Allow-Methods", "GET")
+	r.HandleFunc("/pokemon-upload", uploadPokemon.PokemonUploadHandler).Methods("POST")
+	r.HandleFunc("/pokemon-search/{id}", searchPokemon.PokemonSearchHandler).Methods("GET")
```

Comment:
These already exist as constants in the http package, worth just re-using them.
```suggestion
	r.HandleFunc("/pokemon-search/{id}", searchPokemon.PokemonSearchHandler).Methods(http.MethodGet)
```